db/controller.js

```diff
@@ -14,4 +14,25 @@ module.exports.getArticleById = function(req, res) {
     if (rows.length) res.status(200).send({article: rows[0]})
     else res.status(404).send({msg: "No such article"})
   })
+}
+
+module.exports.getArticles = function(req, res) {
+  model.fetchArticles()
+  .then(({rows}) => {
+    const articlesById = {}
+    rows.forEach((entry) => {
+      delete entry.body
+      articlesById[entry.article_id] = entry
+    })
+    const commentLists = []
+    for (let i in articlesById) {
+      commentLists[i] = model.fetchCommentsByArticle(i)
+    }
+    Promise.all(commentLists).then((commentListsRes) => {
+      for (let i in articlesById) {
+        articlesById[i].comment_count = commentListsRes[i].rows.length
+      }
```

Comment:
This works, but it should not be done here and ideally should be dont by your query and in the model file